src/image/morphology/erode.js

```diff
@@ -101,3 +114,45 @@ function erodeOnceBinary(img, kernel) {
   return newImage;
 }
 
+function erodeOnceBinaryOnlyOnes(img, newImage, kernelWidth, kernelHeight) {
+  const radiusX = (kernelWidth - 1) / 2;
+  const radiusY = (kernelHeight - 1) / 2;
+
+  const minList = [];
+  for (let x = 0; x < img.width; x++) {
+    minList.push(0);
+  }
+
+  for (let y = 0; y < img.height; y++) {
+    let start = y - radiusY;
+    for (let x = 0; x < img.width; x++) {
+      let min = 1;
+      for (let h = 0; h < kernelHeight; h++) {
+        const hh = start + h;
+        if (hh < 0 || hh >= img.height) continue;
+        if (img.getBitXY(x, hh) === 0) {
+          min = 0;
+          break;
+        }
+      }
+      minList[x] = min;
+    }
+
+    for (let x = 0; x < img.width; x++) {
+      let min = 1;
+      for (let ii = 0; ii < kernelWidth; ii++) {
+        let i = ii - radiusX + x;
+        if (i < 0 || i >= img.width) continue;
+        if (minList[i] === 0) {
+          min = 0;
+          break;
+        }
+      }
+
```

Comment:
Can we made assumption about the newImage ? Is it a copy of the previous image ?
Because if the currentBit is `0` we even don't have to analyse this position and should do nothing.
It is only in the case it is `1` and there is any `0` in the minList that we should set to '0' and break.
Seems to me the variable `min` is useless.

Reply:
newImage is a ... new image! But you are right. We can read the current value at the beginning. I will make a change.

Reply:
It is not faster (tested with a 3000x2000 image and 50x50 kernel), speed is the same. Do you still prefer if I change it?